alf/layers.py

```diff
@@ -424,6 +424,143 @@ def make_parallel(self, n):
         return ParallelFC(n=n, **self._kwargs)
 
 
+class FCBatchEnsemble(FC):
+    r"""The BatchEnsemble for FC layer.
+
+    BatchEnsemble is proposed in `Wen et. al. BatchEnsemble: An Alternative Approach
+    to Efficient Ensemble and Lifelong Learning <https://arxiv.org/abs/2002.06715>`_
+
+    In a nutshell, a tuple of vector :math:`(r_k, s_k)` is maintained for ensemble
+    member k in addition to the original FC weight matrix w. For input x, the
+    result for ensemble member k is calculated as :math:`(W \circ (s_k r_k^T)) x`.
+    This can be more efficiently calculated as :math:`(W (x \circ r_k)) \circ s_k`
```

Comment:
Should add an explanation that every forward will use a random k if no ensemble id is provided? Otherwise one might think that there will be K outputs for each input.